ui/src/org/pentaho/di/ui/util/EnvironmentUtils.java

```diff
@@ -0,0 +1,214 @@
+      return true;
+    }
+    return checkUserAgent( MSIE_PATTERN.matcher( userAgent ), getSupportedVersion( "min.explorer.supported" )  )
+      || checkUserAgent( SAFARI_PATTERN.matcher( userAgent ), getSupportedVersion( "min.safari.supported" )  );
+  }
+
+  private boolean checkUserAgent( Matcher matcher, int version ) {
+    if ( matcher.find() && Integer.parseInt( matcher.group( 1 ) ) < version ) {
+      return true;
+    }
+    return false;
+  }
+
+  /**
+   * getWebkitPath
+   *
+   * Ask for user Agent of the available browser.
+   *
+   * @return a string that contains the user agent of the browser.
+   */
+  protected String getUserAgent() {
+    Browser browser;
+    try {
+      browser = new Browser(  new Shell(), SWT.NONE );
+    } catch ( SWTError e ) {
+      e.printStackTrace();
+      return  "";
+    }
+    String userAgent = browser.evaluate( "return window.navigator.userAgent;" ).toString();
+    browser.close();
+    return userAgent;
+  }
+
+  /**
+   * isWebkitUnavailable
+   *
+   * Checks the existence of the webkit library on ubuntu 16 or ubuntu 14 .
+   *
+   * @return 'true' if the webkit library is not present in ubuntu 16 or ubuntu 14 , 'false' otherwise.
+   */
+  public synchronized boolean isWebkitUnavailable() {
+    String path = getWebkitPath();
+    String osName = getEnvironmentName();
+    if ( ( path == null || path.length() < 1 || !path.contains( "webkit" ) )
+        &&
+        ( osName.contains( SUPPORTED_DISTRIBUTION_NAME + " " + getSupportedVersion( "max.ubuntu.supported" ) )
+            ||
+            osName.contains( SUPPORTED_DISTRIBUTION_NAME + " " + getSupportedVersion( "min.ubuntu.supported" ) ) ) ) {
+      return true;
+    }
+    return false;
+  }
+
+  /**
+   * getWebkitPath
+   *
+   * Ask for the path in the system for the webkit library.
+   *
+   * @return a string that contains the path or 'null' if not found.
+   */
+  protected String getWebkitPath() {
+    return System.getenv( "LIBWEBKITGTK" );
+  }
+
+  /**
+   * getOsName
+   *
+   * Ask for the Operating system name.
+   *
+   * @return a string that contains the current Operating System.
+   */
+  private String getEnvironmentName() {
+    String osName = getOsName();
+    if ( osName.contentEquals( "linux" ) ) {
+      return osName + " " + getDistro().toLowerCase();
+    }
+    return osName;
+  }
+
+  protected String getOsName() {
+    return System.getProperty( "os.name" ).toLowerCase();
+  }
+
+  /**
+   *
+   * @param property
+   * @return the value of the requiredProperty
+   */
+  protected int getSupportedVersion( String property ) {
+    return PropsUI.getInstance().getSupportedVersion( property );
+  }
+
+  /**
+   * getDistro
+   *
+   * Ask for the running linux distribution.
+   *
+   * @return a string that contains the distribution name.
+   */
+  private String getDistro() {
+    Process p = null;
+    try {
+      p = ExecuteLine();
+    } catch ( IOException e ) {
+      e.printStackTrace();
```

Comment:
we can continue working just returning an empty string is not critical, ill add the logger